data/database.py

```diff
@@ -1,58 +1,177 @@
+        the year, month, day, hour, minute and second of the moment
+        """
+        i = datetime.datetime.now()
+        db_name = str(i.year) + str(i.month) + str(i.day) + str(i.hour) \
+            + str(i.minute) + str(i.second)
+        while self.db_exist(db_name):
+            db_name = str(int(db_name) + 1)
+        return db_name
+
+    def run_migrations(self, c):
+        """This function creates table for room, person and livingspace"""
+        c.execute("CREATE TABLE room_table (name TEXT PRIMARY KEY, \
+                                        type TEXT, capacity INTEGER)")
+
+        c.execute("CREATE TABLE person_table (id TEXT PRIMARY KEY,\
+                            name TEXT, designation TEXT, office TEXT)")
+
+        c.execute("CREATE TABLE livingspace_table (ids TEXT PRIMARY KEY,\
+                            wants_accommodation TEXT, livingspace TEXT)")
+
+    def load_state(self, db_name):
+        """
+        This function loads application data from the database
+        and returns it to the application
+        """
+        sqlite_file = "data/{}.sqlite".format(db_name)
+        conn = sqlite3.connect(sqlite_file)
+        c = conn.cursor()
+
+        c.execute("select name, type from room_table")
+        db_rooms = c.fetchall()
+        rooms = self.get_rooms_from_db_rooms(db_rooms)
+
+        c.execute("select id, name, designation, office," +
+                  " wants_accommodation, livingspace from " +
+                  "person_table INNER JOIN " +
+                  "livingspace_table ON id = " +
+                  "ids WHERE designation = 'fellow'")
+        db_fellow_list = c.fetchall()
+        fellow_list = self.get_fellows_from_db_fellows(db_fellow_list, rooms)
+
+        c.execute(
+            "select id, name, designation, office" +
+            " from person_table WHERE designation = 'staff'")
+        db_staff_list = c.fetchall()
+        staff_list = self.get_staffs_from_db_staffs(db_staff_list, rooms)
+        app_data = {"all_rooms": rooms, "staff_list": staff_list,
+                    "fellow_list": fellow_list}
+        conn.commit()
+        conn.close()
+        return app_data
+
+    def get_rooms_from_db_rooms(self, db_rooms):
+        """
+        This function extracts the collection of rooms from the
+        rows of room returned from the database
+        """
+        rooms = []
+        for row in db_rooms:
+            new_room = Office(row[0]) if row[1].lower() == \
+                    "office" else LivingSpace(row[0])
+            rooms.append(new_room)
+        return rooms
+
+    def get_fellows_from_db_fellows(self, db_fellow_list, rooms):
+        """
+        This function extracts the collection of rooms from the
+        rows of room returned from the database
+        """
+        fellow_list = []
+        for row in db_fellow_list:
+            new_fellow = Fellow(row[1], row[2])
+            new_fellow.ID = row[0]
```

Comment:
On creating an instance of fellow, no ID is assigned until the calling of the function generate_id but in this case, the ID been retrieved from the database.